src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -430,52 +434,72 @@ public void Initialize()
+                    functionMetadata = ReadFunctionMetadata(ScriptConfig, TraceWriter, _startupLogger, FunctionErrors, _settingsManager);
+                    TraceWriter.Verbose("Function metadata read.");
+                }
+
                 var extensionLoader = new ExtensionLoader(ScriptConfig, TraceWriter, _startupLogger);
                 var usedBindingTypes = extensionLoader.DiscoverBindingTypes(functionMetadata);
 
                 var bindingProviders = LoadBindingProviders(ScriptConfig, hostConfigObject, TraceWriter, _startupLogger, usedBindingTypes);
                 ScriptConfig.BindingProviders = bindingProviders;
+                TraceWriter.Verbose("Binding providers loaded.");
 
                 var coreBinder = bindingProviders.OfType<CoreExtensionsScriptBindingProvider>().First();
                 coreBinder.AppDirectory = ScriptConfig.RootScriptPath;
 
                 // Allow BindingProviders to initialize
-                foreach (var bindingProvider in ScriptConfig.BindingProviders)
+                using (metricsLogger.LatencyEvent(MetricEventNames.HostStartupInitializeBindingProvidersLatency))
                 {
-                    try
+                    foreach (var bindingProvider in ScriptConfig.BindingProviders)
                     {
-                        bindingProvider.Initialize();
-                    }
-                    catch (Exception ex)
-                    {
-                        // If we're unable to initialize a binding provider for any reason, log the error
-                        // and continue
-                        string errorMsg = string.Format("Error initializing binding provider '{0}'", bindingProvider.GetType().FullName);
-                        TraceWriter.Error(errorMsg, ex);
-                        _startupLogger?.LogError(0, ex, errorMsg);
+                        try
+                        {
+                            bindingProvider.Initialize();
+                        }
+                        catch (Exception ex)
+                        {
+                            // If we're unable to initialize a binding provider for any reason, log the error
+                            // and continue
+                            string errorMsg = string.Format("Error initializing binding provider '{0}'", bindingProvider.GetType().FullName);
+                            TraceWriter.Error(errorMsg, ex);
+                            _startupLogger?.LogError(0, ex, errorMsg);
+                        }
                     }
+                    TraceWriter.Verbose("Binding providers initialized.");
                 }
+
                 extensionLoader.LoadBuiltinExtensions(usedBindingTypes);
 
                 var directTypes = GetDirectTypes(functionMetadata);
                 extensionLoader.LoadDirectlyReferencedExtensions(directTypes);
                 extensionLoader.LoadCustomExtensions();
+                TraceWriter.Verbose("Extension loading complete.");
```

Comment:
Don't we already have a trace for each extension loaded, so the last one loaded gives us the complete timestamp?

Reply:
Nope. There's ~600ms here:

```
2018-02-22 17:00:12.7935098 Loaded custom extension 'EventGridExtensionConfig'
2018-02-22 17:00:12.8521540 Extension loading complete.
```